main.nf

```diff
@@ -406,6 +408,35 @@ process generate_decoy_database {
 //     """
 //}
 
+process openms_peakpicker {
+
+    label 'process_low'
+
+    publishDir "${params.outdir}/logs", mode: 'copy', pattern: '*.log'
+
+    input:
+     tuple mzml_id, path(mzml_file) from mzmls_pp
+
+    when:
+      params.openms_peakpicking
+
+    output:
+     set mzml_id, file("${mzml_file.baseName}_picked.mzML") into mzmls_comet, mzmls_msgf, mzmls_plfq
+     file "*.log"
+
+    script:
+     // TODO maybe allow specifying ms-levels and inmemory
+     """
+     PeakPickerHiRes -in ${mzml_file} \\
+                     -out ${mzml_file.baseName}_picked.mzML \\
+                     -threads ${task.cpus} \\
+                     -debug ${params.pp_debug} \\
+                     -processOption lowmemory \\
```

Comment:
does the low memory also detect if it is picked?
the in memory has the "auto" mode

Reply:
it uses PeakPickerHiRes.pick()
https://github.com/OpenMS/OpenMS/blob/develop/src/topp/PeakPickerHiRes.cpp#L149

Reply:
no it is not:
https://github.com/OpenMS/OpenMS/blob/develop/src/openms/source/TRANSFORMATIONS/RAW2PEAK/PeakPickerHiRes.cpp#L525
Why is this if-case not inside the pick function?

Reply:
hmm maybe because of the templated code?
Should still be possible to put it in the pick for MSSpectrum

Reply:
ok but anyway. nothing we can do here except allowing inmemory. which I did.